hnn_core/network.py

```diff
@@ -750,6 +750,33 @@ def _instantiate_drives(self, n_trials=1):
                 self.external_drives[
                     drive['name']]['events'].append(event_times)
 
+    def connect_cells(self, src_gids, target_gids, weight, receptor, location,
+                      delay, space_constant):
+        """Pass connectivity matrix defining cell-cell connections.
+
+        Parameters
+        ----------
+        src_gids: list of int
+            List of gid sources where axon originates.
+        target_gids: list of int
+            List of gid targets where axon terminates.
+        weight: np.array (n_src_gids, n_target_gids) of float
+            Array defining synaptic weights.
+        receptor: np.array (n_src_gids, n_target_gids) of str
+            Array defining recptors on target cell. Must be one of
+            'ampa', 'nmda', 'gabaa', or 'gabab'.
+        location: np.array (n_src_gids, n_target_gids) of str
+            Array defining location of synapse on target cell. Must be one of
+            'proximal' or 'distal'.
+        delay: np.array (n_src_gids, n_target_gids) of float
+            Array defining synaptic delays.
+        space_constant: np.array (n_src_gids, n_target_gids) of float
+            Array defining space constants.
```

Comment:
Could we start simple and add more features as the need arises? Maybe we start with just binary matrices specifying whether a connection is on or off?

Reply:
I think this definitely makes sense to start, but it will be pretty trivial to add in this functionality once the groundwork is in place for defining connections. Especially if we go with the list of tuples format since it will just come down to iterative calls to `_connect_celltypes()`.